resolver_wrapper.go

```diff
@@ -194,5 +194,7 @@ func (ccr *ccResolverWrapper) addChannelzTraceEvent(s resolver.State) {
 	} else if len(ccr.curState.Addresses) == 0 && len(s.Addresses) > 0 {
 		updates = append(updates, "resolver returned new addresses")
 	}
-	channelz.Infof(logger, ccr.cc.channelz, "Resolver state updated: %s (%v)", pretty.ToJSON(s), strings.Join(updates, "; "))
+	if channelz.IsOn() || logger.V(2) {
```

Comment:
What is the reason for the additional channelz.IsOn() bool conditional?

Reply:
It's similar to the reason for the `logger.V(2)` check. We only want to call `pretty.ToJSON` when we know we're going to use the result. `channelz.Infof` can add a trace event, log, or both, so we want to call it when channelz data collection is on, we're logging at depth 2, or both.